front/app/containers/Admin/reporting/components/ReportBuilder/Widgets/SurveyResultsWidget/SurveyQuestionFilter.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { ChangeEvent } from 'react';
+
+// hooks
+import useProject from 'hooks/useProject';
+import useFormResults from 'hooks/useFormResults';
+import useLocalize from 'hooks/useLocalize';
+
+// utils
+import { useIntl } from 'utils/cl-intl';
+import { isNilOrError } from 'utils/helperUtils';
+import { Box, Checkbox, Text } from '@citizenlab/cl2-component-library';
+
+// messages
+import messages from './messages';
+
+type SurveyQuestionFilterProps = {
+  projectId: string;
+  phaseId: string;
+  showQuestions: number[] | undefined;
+  onToggleQuestion: (showQuestions: number[]) => void;
+};
+
+const SurveyQuestionFilter = ({
+  projectId,
+  phaseId,
+  showQuestions,
+  onToggleQuestion,
+}: SurveyQuestionFilterProps) => {
+  const localize = useLocalize();
+  const { formatMessage } = useIntl();
+  const project = useProject({ projectId });
+  const formResults = useFormResults({
+    projectId,
+    phaseId,
+  });
+
+  if (isNilOrError(formResults) || isNilOrError(project)) {
+    // This never seems to get shown
+    return (
+      <Text variant="bodyM" color="textSecondary">
+        {formatMessage(messages.surveyNoQuestions)}
+      </Text>
+    );
+  }
+
+  const { results } = formResults;
+  if (results.length === 0) {
+    return null;
+  }
+
+  // Initialise which questions are checked
+  let selectedQuestions: number[] = [];
+  if (showQuestions !== undefined) {
+    selectedQuestions = showQuestions;
+  } else {
+    for (let i = 0; i <= results.length - 1; i++) {
+      selectedQuestions.push(i);
+    }
+    onToggleQuestion(selectedQuestions);
```

Comment:
This is probably the cause of the 'cannot update component' bug: you shouldn't call setter functions like this in the top-level of the component, because it's going to call this function on every re-render. It should probably only happen once, or as a result of user action